driver/src/orderbook/streamed/updating_orderbook.rs

```diff
@@ -77,28 +79,23 @@ impl UpdatingOrderbook {
         };
     }
 
-    /// Use the context, ensuring that the orderbook has been initialized.
-    fn do_with_context<T>(&self, callback: impl FnOnce(&mut Context) -> Result<T>) -> Result<T> {
-        // We unwrap because there is no way the mutex can be poisoned. If the mutex somehow was
-        // poisoned anyway against our expectation then panicking is correct because we cannot
-        // recover from this.
-        // The mutex cannot be poisoned because `do_with_context` is the only function accessing
-        // the mutex and is only called within this file. The `callback`s we use do not panic.
-        let mut context_guard = self.context.lock().expect("poison error");
+    /// Ensure that the orderbook has been initialized and updated.
+    /// Post condition on success: self.context is Some(_).
+    async fn update_context(&self) -> Result<()> {
+        let mut context_guard = self.context.lock().await;
         let context_optional: &mut Option<Context> = &mut context_guard;
         match context_optional {
```

Comment:
ubernit:
```suggestion
        match context_guard.as_mut() {
```

Reply:
Doesn't work because we're writing to `*context_optional` below.

Reply:
`MutexGuard` implements `DerefMut` só you can write to it directly.